library_conventions/packages.adoc

```diff
@@ -0,0 +1,232 @@
+* *Size information must be given in millimeters*, not inches or mils. For
+  packages which are well known by their size in inches (e.g. "0805" which is
+  "2012" in metric), it's recommended to write the well known name in
+  parentheses. For example a resistor could be called "RESC2012X70 (0805)".
+* For packages which are available with different pin counts, *append the pin
+  count with a hyphen as separator and omit leading zeros* (e.g. "DIP-8" instead
+  of "DIP08").
+* For manufacturer-specific packages, start with the manufacturers name
+  (e.g. "Molex 53261-06").
+
+
+[#libraryconventions-packages-pads]
+==== Pads
+
+* *Always add all pads of Packages*, even those which are not connected in
+  some parts! For example the package "TO220" has a metal tab, so you should
+  define it as a pad, no matter if it's often not connected (and even not
+  connectable when mounted vertically).
+* *Use pad names according {ipc7351-pdf}[IPC-7351]* (if applicable). For
+  Packages which are not covered by IPC-7351:
+** If the function of a pad is absolutely clear, choose its abbreviated
+   functionality as name (e.g. "A" for anode and "C" for cathode).
+** Otherwise just use numbers starting with "1" at top left and increment
+   them counterclockwise.
+
+
+[#libraryconventions-packages-footprints]
+==== Footprint Variants
+
+Within a Package there can be multiple footprint variants. They are intended
+to support following use-cases:
+
+* *Mounting variants*: For example a THT resistor can be mount either
+  vertically or horizontally with various pad distance. Every mounting
+  variant should be available as footprint variants.
+* *Soldering techniques*: Many packages can be soldered either by reflow-,
+  wave- or hand-soldering, which usually require different land patterns. For
+  every suitable soldering technique there should be a corresponding footprint
+  variant.
+* *Density levels*: {ipc7351-pdf}[IPC-7351] specifies three different density
+  levels for footprints:
++
+--
+* Density Level A: Maximum (Most) Land Protrusion
+* Density Level B: Median (Nominal) Land Protrusion
+* Density Level C: Minimum (Least) Land Protrusion
+--
+If applicable, these three density levels should also be added as footprint
+variants.
+
+.Combinations
+[NOTE]
+====
+As a given Package might support multiple of the use-cases above, all suitable
+combinations of them should be added. For example a Package which should have
+all three density levels as defined in IPC-7351 and can be mount either
+vertically or horizontally would need six footprint variants to support all
+possible use-cases.
+====
+
+.Set default footprint
+[IMPORTANT]
+====
+*The first footprint is always the default footprint*, so you should move
+the most reasonable footprint to the top of the footprint list! The default
+footprint should fulfill these rules:
+
+* Designed according density level A (if applicable)
+* Suitable for reflow soldering (if applicable)
```

Comment:
But Level A is always suitable for reflow soldering too, isn't it? I guess it's suitable for all of the most commonly used soldering techniques, reflow, wave and hand soldering. Thus IMHO it's the best choice for the default footprint.

Reply:
I'm not sure. My feeling is that level B (median / nominal protrusion) should probably be the best "average case" size. IPC explicitly lists "wave soldering" as a use case for level A:

![img](https://tmp.dbrgn.ch/screenshots/20181218214908-4wanx254.png)

Wave soldering footprints may be quite large and may also contain things like "solder thieves", see https://electronics.stackexchange.com/a/37706/161033. Even though those can be reflow soldered, it's probably not ideal.

If a package has a reflow soldering footprint (level B) and a wave soldering footprint (level A), and both *could* be soldered in a reflow oven, then according to the guidelines above the wave soldering footprint would be the default. But I'd assume that most users of the software will probably do reflow soldering, since that's the most common soldering/assembly type nowadays. (I read in some places that some companies have completely stopped using wave soldering machines and now even solder THT components using reflow techniques.) Furthermore, I'd estimate that most people will probably tend to do dense designs, even if it's just to save production costs for PCB manufacturing. In both cases those users would have to switch away from the default for every part they're placing.

I'd optimize for the average case, and I suspect that will be the median sized pads.

Reply:
Maybe @HBadertscher has some experience in this area?

Reply:
On the other hand, at least for the chip resistors, the protrusion for level A does not look excessive...

Reply:
> But Level A is always suitable for reflow soldering too, isn't it?

I'm not really sure about that - using Level A pads will lead to more solder paste being applied, which I think could lead to bad solder joints. But I haven't built mass-produced PCBs for some time, so my memory/intuition may be wrong here.

I would ignore wave soldering for this decision: if you build a PCB which is mass-produced using wave-soldering, then you'll have to invest some time into making your PCB wave-soldering-compatible anyways.

I'd go with the footprint that matches manufacturer's recommendation most closely as the default. I think this is around Level B, so I'd go for that (and mention to follow manufacturer's recommendation?)

Reply:
> But I'd assume that most users of the software will probably do reflow soldering, since that's the most common soldering/assembly type nowadays.

For automated assembly I agree, but many PCBs (especially of hobbyists, but also for prototypes in industry) are soldered by hand. Thus I would like to have a default footprint with pads large enough to also allow hand-soldering.

But probably(/hopefully) density level B is suitable for hand soldering too, at least for simple components like resistors/capacitors and many IC packages...

> using Level A pads will lead to more solder paste being applied, which I think could lead to bad solder joints.

Good point.

> I'd go with the footprint that matches manufacturer's recommendation most closely as the default. I think this is around Level B, so I'd go for that

OK, let's recommend density level B as default. I updated the documentation accordingly.

> and mention to follow manufacturer's recommendation?

I added a sentence to recommend the footprint from the datasheet as default footprint of manufacturer-specific packages.